src/performance.js

```diff
@@ -0,0 +1,101 @@
+      l.domainLookupStart = ntEntry.domainLookupStart;
+      l.domainLooupTime   = ntEntry.domainLookupEnd - ntEntry.domainLookupStart;
+      l.connectStart  = ntEntry.connectStart;
+      l.connectTime   = ntEntry.connectEnd - ntEntry.connectStart;
+      l.requestStart  = ntEntry.requestStart;
+      l.requestTime   = ntEntry.responseStart - ntEntry.requestStart;
+      l.responseStart = ntEntry.responseStart;
+      l.responseTime  = ntEntry.responseEnd - ntEntry.responseStart;
+      l.domInteractive= ntEntry.domInteractive;
+      l.domContentLoadedEventStart = ntEntry.domContentLoadedEventStart;
+      l.domContentLoadedEventTime  =
+        ntEntry.domContentLoadedEventEnd - ntEntry.domContentLoadedEventStart;
+      l.domComplete   = ntEntry.domComplete;
+      l.loadEventStart= ntEntry.loadEventStart;
+      l.loadEventTime = ntEntry.loadEventEnd - ntEntry.loadEventStart;
+
+      l.totalTime     = ntEntry.loadEventEnd - l.startTime;
+    } else {
+      l.redirectCount = performance.navigation.redirectCount;
+      l.startTime     =
+        timing.redirectStart === 0 ? timing.fetchStart:timing.redirectStart;
+
+      // timing.unloadEventStart maybe larger than l.startTime
+      l.unloadEventStart = timing.unloadEventStart;
+
+      l.unloadEventTime  = timing.unloadEventEnd - timing.unloadEventStart;
+
+      l.redirectStart = timing.redirectStart;
+      l.redirectTime  = timing.redirectEnd - timing.redirectStart;
+      l.fetchStart    = timing.fetchStart - l.startTime;
+      l.domainLookupStart = timing.domainLookupStart - l.startTime;
+      l.domainLooupTime   = timing.domainLookupEnd - timing.domainLookupStart;
+      l.connectStart  = timing.connectStart - l.startTime;
+      l.connectTime   = timing.connectEnd - timing.connectStart;
+      l.requestStart  = timing.requestStart - l.startTime;
+      l.requestTime   = timing.responseStart - timing.requestStart;
+      l.responseStart = timing.responseStart - l.startTime;
+      l.responseTime  = timing.responseEnd - timing.responseStart;
+      l.domInteractive= timing.domInteractive - l.startTime;
+      l.domContentLoadedEventStart =
+        timing.domContentLoadedEventStart - l.startTime;
+      l.domContentLoadedEventTime  =
+        timing.domContentLoadedEventEnd - timing.domContentLoadedEventStart;
+      l.domComplete   = timing.domComplete - l.startTime;
+      l.loadEventStart= timing.loadEventStart - l.startTime;
+      l.loadEventTime = timing.loadEventEnd - timing.loadEventStart;
+
+      l.totalTime     = timing.loadEventEnd - l.startTime;
+    }
+
+    l.domLoading = timing.domLoading - (
+      timing.redirectStart === 0 ? timing.fetchStart:timing.redirectStart);
+    l.domComplete = timing.domComplete - timing.domLoading;
+
+    for (var k in l) {
+      l[k] = Math.round(l[k]);
+    }
+
+    return l;
+  };
+
+  function startCollect() {
+    setTimeout(function() {
+      const l = LoadPerformance();
+      if (l.loadEventTime > 0) {
```

Comment:
I did some testing and found one issue: on some websites counter was not shown in the toolbar. I debugged it and found the problem with the condition on this line. So you do the check on `loadEventTime` which is `loadEventEnd - loadEventStart` and the problem is when these two times are very close. For example on twitter.com in my testing it is just 0.1 msec between these two events. Since you round them to a nearest ms this differences turns into 0 so it doesn't enter the code block. I suggest to change this check to use `l.totalTime` instead, this fixed the problem for me.

Reply:
You said it was very useful. I really did not consider this issue.